cmta/src/main/scala/com/lunatech/cmt/admin/command/New.scala

```diff
@@ -0,0 +1,137 @@
+import com.lunatech.cmt.client.Configuration
+import com.lunatech.cmt.client.cli.CmtcCommand
+import com.lunatech.cmt.Domain.InstallationSource.*
+import com.lunatech.cmt.Helpers.ignoreProcessStdOutStdErr
+import sbt.io.IO as sbtio
+import sbt.io.syntax.*
+import com.lunatech.cmt.*
+
+import cats.syntax.either.*
+
+import java.io.FileFilter
+import sys.process.*
+import util.{Try, Success, Failure}
+
+object New:
+
+  private final case class TagSet(tags: Vector[String])
+
+  @AppName("new")
+  @CommandName("new")
+  @HelpMessage(
+    "Create a new course from an existing course template in a Github repository - by default the `lunatech-labs` organisation is used.")
+  final case class Options(
+      @ExtraName("t")
+      @ValueDescription(
+        "the template course to use - provide in the format 'organisation/project' or just 'project' if the project is in the lunatech-labs organisation on Github")
+      template: CourseTemplate,
+      @ExtraName("c")
+      @ValueDescription("The (optional) configuration file to use during processing of the command")
+      @HelpMessage(
+        "if not specified will default to the config file present in the directory provided by the --main-repository argument")
+      maybeConfigFile: Option[ConfigurationFile] = None)
+
+  given Validatable[New.Options] with
+    extension (options: New.Options)
+      def validated(): Either[CmtError, New.Options] =
+        Right(options)
+  end given
+
+  given Executable[New.Options] with
+    extension (options: New.Options)
+      def execute(configuration: Configuration): Either[CmtError, String] =
+        for {
+          // list the contents of the ~/Courses directory, if there's anything already matching the
+          // name then get the count so we can append to the name and prevent conflicts
+          template <- options.template.value
+          existingFilesWithSameName = sbtio.listFiles(
+            configuration.coursesDirectory.value,
+            new FileFilter {
+              override def accept(file: File): Boolean =
+                file.name.startsWith(template.project)
+            })
+          discriminator = if (existingFilesWithSameName.size > 0) s"-${existingFilesWithSameName.size}" else ""
+          targetDirectoryName = s"${template.project}$discriminator"
+          newRepo <- newCmtRepoFromGithubProject(template, targetDirectoryName, configuration)
+        } yield newRepo
+
+      private def cloneMainRepo(githubProject: GithubProject, tmpDir: File): Either[CmtError, TagSet] =
+        val project = githubProject.project
+        val organisation = githubProject.organisation
+        val tag = githubProject.tag
+        val cloneGit = Process(Seq("git", "clone", s"git@github.com:$organisation/$project.git"), tmpDir)
+        val cloneGh = Process(Seq("gh", "repo", "clone", s"$organisation/$project"), tmpDir)
+        // TODO CHECK: I wonder if we should support cloning via HTTP...
```

Comment:
I don't see why we shouldn't - i think there are many people NOT using ssh so this would be necessary for them to use CMT

Reply:
Agreed; There's corner case where the clone is done from a private repo. In that case, cloning via HTTP triggers a request from git for user credentials (via stdin/stdout). Then git reports that support for password authentication on HTTP is no longer supported. In any case, it should work without a problem for public repo's which covers a large chunk of the use cases. Also, the order in which the cloning is tried is ssh, gh (command), and http.
I'll run a few more experiments to figure out more details.